battlesnake-rs/src/mcts_snake.rs

```diff
@@ -0,0 +1,442 @@
+
+        if game_info.ruleset.name == "wrapped" {
+            let game = WrappedCellBoard4Snakes11x11::convert_from_game(game, &id_map).unwrap();
+
+            let snake = MctsSnake::new(game, game_info);
+
+            Box::new(snake)
+        } else {
+            let game = StandardCellBoard4Snakes11x11::convert_from_game(game, &id_map).unwrap();
+
+            let snake = MctsSnake::new(game, game_info);
+
+            Box::new(snake)
+        }
+    }
+
+    fn about(&self) -> AboutMe {
+        AboutMe {
+            author: Some("coreyja".to_owned()),
+            color: Some("#fc0398".to_owned()),
+            head: Some("trans-rights-scarf".to_owned()),
+            ..Default::default()
+        }
+    }
+}
+
+impl<
+        T: Clone
+            + SimulableGame<Instrument, 4>
+            + PartialEq
+            + RandomReasonableMovesGame
+            + VictorDeterminableGame
+            + YouDeterminableGame
+            + 'static,
+    > BattlesnakeAI for MctsSnake<T>
+{
+    #[tracing::instrument(
+        level = "info",
+        skip_all,
+        fields(total_number_of_iterations, total_score,)
+    )]
+    fn make_move(&self) -> Result<MoveOutput> {
+        let current_span = tracing::Span::current();
+
+        let arena = Arena::new();
+
+        let mut rng = rand::thread_rng();
+
+        let root_node: &mut Node<T> = arena.alloc(Node::new(self.game.clone()));
+
+        root_node.expand(&arena);
+
+        let mut total_number_of_iterations = 0;
+
+        let start = std::time::Instant::now();
+
+        const NETWORK_LATENCY_PADDING: i64 = 100;
+        let max_duration = self.game_info.timeout - NETWORK_LATENCY_PADDING;
+
+        while start.elapsed().as_millis() < max_duration.try_into().unwrap() {
+            total_number_of_iterations += 1;
+
+            let mut next_leaf_node = root_node.next_leaf_node(total_number_of_iterations);
+
+            next_leaf_node = {
+                let borrowed = next_leaf_node;
+
+                // If next_leaf_node HAS been visited, then we expand it
+                if borrowed.number_of_visits.load(Ordering::Relaxed) > 0 {
+                    borrowed.expand(&arena);
+
+                    borrowed.next_leaf_node(total_number_of_iterations)
+                } else {
+                    next_leaf_node
+                }
+            };
+
+            //Now we do a simulation for this leaf node
+            let score = next_leaf_node.simulate(&mut rng);
+
+            //We now need to backpropagate the score
+            next_leaf_node.backpropagate(score);
+        }
+
+        // We are outside the loop now and need to pick the best move
+        current_span.record("total_number_of_iterations", &total_number_of_iterations);
+        current_span.record(
+            "total_score",
+            &root_node.total_score.load(Ordering::Relaxed),
+        );
+
+        let best_child = root_node.best_child(root_node.number_of_visits.load(Ordering::Relaxed));
```

Comment:
This usage of `best_child` method is wrong

`best_child` is currently defined to use the UCB1 score of the children, and pick the highest. This is the 'best' when we are looking for the next leaf node.
But here we want to child with the highest average score, which this doesn't calculate

This also explains why we need the total number of iterations to determine the `best_child`. We need that for UCB1, but not for average score. That's why it felt weird to pass that in here